framework/decode/vulkan_ascii_consumer_base.h

```diff
@@ -109,22 +111,52 @@ class VulkanAsciiConsumerBase : public VulkanConsumer
 
   protected:
     template <typename ToStringFunctionType>
-    inline void WriteApiCallToFile(const ApiCallInfo&   call_info,
-                                   const std::string&   functionName,
-                                   util::ToStringFlags  toStringFlags,
-                                   uint32_t&            tabCount,
-                                   uint32_t             tabSize,
-                                   ToStringFunctionType toStringFunction)
+    inline void WriteApiCallToFile(const ApiCallInfo& call_info,
+                                   const std::string& functionName,
+                                   /// @todo Remove toStringFlags, tabCount, tabSize
+                                   util::ToStringFlags        toStringFlags,
+                                   const uint32_t&            tabCount,
+                                   const uint32_t             tabSize,
+                                   const ToStringFunctionType toStringFunction,
+                                   /// The formatted return string value excluding trailing comma
+                                   const char* const return_val = nullptr)
     {
-        using namespace util;
-        fprintf(file_, "%s\n", (call_info.index ? "," : ""));
-        fprintf(file_, "\"[%s]%s\":", std::to_string(call_info.index).c_str(), functionName.c_str());
-        fprintf(file_, "%s", GetWhitespaceString(toStringFlags).c_str());
-        fprintf(file_, "%s", ObjectToString(toStringFlags, tabCount, tabSize, toStringFunction).c_str());
+        // Output the start of a function call line, up to the arguments:
+        if (nullptr == return_val)
+        {
+            fprintf(file_,
+                    "{\"index\":%llu,\"vkFunc\":{\"name\":\"%s\",\"args\":{",
+                    (long long unsigned int)call_info.index,
+                    functionName.c_str());
+        }
+        else
+        {
+            fprintf(file_,
+                    "{\"index\":%llu,\"vkFunc\":{\"name\":\"%s\",\"return\":\"%s\",\"args\":{",
```

Comment:
My last commit wrapped the return value in quotation marks in `\"return\":\"%s\",`, which lets the upstream code not bother with the previous string manipulations to put them in place earlier.
This is the spot in which I'm saying in my other comment that the `fprintf` could be pulling the return value chars directly out of our read only data section with some more refactoring.